### PR TITLE
Refactor docker/NewClient to use early return

### PR DESCRIPTION
**Pull request by KeisukeYamashita**
## What

As title.

## Why

For better code. Before `for`-looping, IMO, it is better to check the length and early return if the length is zero.

### CODE REVIEW

File: pkg/auth/docker/client.go [hunk 1]
```diff
@@ -22,13 +22,6 @@ type Client struct {
...
 func NewClient(configPaths ...string) (auth.Client, error) {
 	var configs []*configfile.ConfigFile
-	for _, path := range configPaths {
-		cfg, err := loadConfigFile(path)
-		if err != nil {
-			return nil, errors.Wrap(err, path)
-		}
-		configs = append(configs, cfg)
-	}
 	if len(configs) == 0 {
```
Comment (jdolitsky) [inline pkg/auth/docker/client.go:25]: I think this will always be 0 .. should we check `len(configPaths)` instead ?

Comment (KeisukeYamashita) [reply pkg/auth/docker/client.go]: Oh, sorry! I've fixed it. Thank you for catching this.

File: pkg/auth/docker/client.go [hunk 2]
```diff
@@ -22,22 +22,26 @@ type Client struct {
 // All changes will only be written to the first config file.
 func NewClient(configPaths ...string) (auth.Client, error) {
 	var configs []*configfile.ConfigFile
```
Comment (jdolitsky) [inline pkg/auth/docker/client.go:24]: Once last thing is that this var can be defined after the early return (`configs`)

Comment (KeisukeYamashita) [reply pkg/auth/docker/client.go]: Thank you 🙇

**Comment by KeisukeYamashita** [comment]
@jdolitsky Is it /test ? https://github.com/deislabs/oras/pull/122#issuecomment-531494124

Do I need to do something? I read the "contributions" section but it didn't tell me what to do after submitting a PR.

**Comment by jdolitsky** [comment]
@KeisukeYamashita yes thanks :)

**Comment by jdolitsky** [comment]
cc @shizhMSFT

**Pull request merged**